doc: improve nginx configuration and certbot setup

maethor opened the pull request:
Install certbot before Nginx
Install certbot from stretch-backports
Configure Nginx to manage user agent blacklist

bbeaulant commented:
Nice, thank you.

Few things about your PR : 
- `rewrite ^(.*)$ /maintenance.html last;` on banned_ip clause was not a mistake. This was designed to lure the banned user
- `--email contact@lairdubois.fr` would be preferable than `root@sysmove.fr` ;)
- is `aptitude` better than `apt-get` ?
- Current conf is not exactly the same no ? `include snippets/letsencrypt.conf;` better way ?

maethor commented:
OK, I updated the PR in regard of your 3 firsts comments. The email was obviously a copy/paste mistake from my doc, and `aptitude` is way better than `apt-get` but it is irrelevant here so I replaced it by `apt-get`.

About your last comment, using a snippet is cleaner because it avoids duplicating configuration, but in the context of your standalone documentation it is simplier to put everything in one file than to add a snippet.

bbeaulant commented:
Thank you. (But your force-push was empty. I did changes ;) )
